daemon/network.go

```diff
@@ -560,38 +560,39 @@ func (daemon *Daemon) deleteNetwork(nw *libnetwork.Network, dynamic bool) error
 }
 
 // GetNetworks returns a list of all networks
-func (daemon *Daemon) GetNetworks(filter filters.Args, config types.NetworkListConfig) ([]types.NetworkResource, error) {
-	networks := daemon.getAllNetworks()
-
-	list := make([]types.NetworkResource, 0, len(networks))
+func (daemon *Daemon) GetNetworks(filter filters.Args, config types.NetworkListConfig) (networks []types.NetworkResource, err error) {
 	var idx map[string]*libnetwork.Network
 	if config.Detailed {
 		idx = make(map[string]*libnetwork.Network)
 	}
 
-	for _, n := range networks {
+	allNetworks := daemon.getAllNetworks()
+	networks = make([]types.NetworkResource, 0, len(allNetworks))
+	for _, n := range allNetworks {
 		nr := buildNetworkResource(n)
-		list = append(list, nr)
+		networks = append(networks, nr)
 		if config.Detailed {
 			idx[nr.ID] = n
 		}
 	}
 
-	var err error
-	list, err = internalnetwork.FilterNetworks(list, filter)
+	networks, err = internalnetwork.FilterNetworks(networks, filter)
 	if err != nil {
 		return nil, err
 	}
 
 	if config.Detailed {
-		for i := range list {
-			np := &list[i]
-			buildDetailedNetworkResources(np, idx[np.ID], config.Verbose)
-			list[i] = *np
+		for i := range networks {
+			np := &networks[i]
+			np.Containers = buildContainerAttachments(idx[np.ID])
+			if config.Verbose {
```

Comment:
I'm wondering why we have this `Verbose` flag in the first place :thinking: On the CLI, the help message says: `Verbose output for diagnostics`. That's pretty much unhelpful.

Looks like it was introduced in #31710 and the intent is to show services running on other Swarm nodes. I hoped for years this option existed and I'm only learning about it today :upside_down_face: We should either:

* Rename it to `Services` / `--services` ;
* Return it unconditionally when the network is an overlay ;

Also, this flag is only available through the `getNetwork` route. This code should probably be moved to a dedicated backend method `GetNetwork` (instead of relying on `GetNetworks` to get a single network).

Reply:
Ah, yes, that used to be the "default".. but it wasn't, ahem, great if you used that on an actual cluster with (say) 300 nodes.. and now the API started to return 1000 service attachments for networks